types/hapi__call/index.d.ts

```diff
@@ -1,29 +1,32 @@
-// Type definitions for @hapi/call 8.0
+// Type definitions for @hapi/call 9.0
 // Project: https://github.com/hapijs/call#readme
 // Definitions by: Rodrigo Saboya <https://github.com/saboya>
+//                 Sebastian Malton <https://github.com/nokel81>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
 /// <reference types="node" />
 
+export type RouteMethod = "get" | "head" | "post" | "put" | "delete" | "connect" | "options" | "trace" | "patch";
+
 export interface RouteDefinition {
-    method: string;
+    method: RouteMethod | "*";
     path: string;
 }
 
 export interface RouterOptions {
     isCaseSensitive: boolean;
 }
 
-export interface Match<P = any, D = any> {
-    params: P;
+export interface Match<Handler> {
+    params: Partial<Record<string, string>>;
     paramsArray: string[];
-    route: D;
+    route: Handler;
 }
 
-export type Route<P = any, D = any> = Match<P, D> | Error;
+export type Route<Handler> = Match<Handler> | Error;
 
-export class Router {
+export class Router<Handler> {
     constructor(routerOptions?: RouterOptions)
-    add(definition: RouteDefinition, data?: any): void;
-    route(method: string, path: string): Route;
+    add(definition: RouteDefinition, route?: Handler): void;
+    route(method: RouteMethod, path: string): Route<Handler>;
```

Comment:
Can you explain your motivation for these changes?

Reply:
- What does `Handler` actually represent?
- Why is its type applied at `new Router()` instead of `router.route()`?
- Why did you remove the ability to describe `params` instead of just constraining it?

Reply:
3. Because it not really possible to constrain it. None of the methods provide type parameters that you could pass in (not that you should because that is just hiding a cast). The only use for `Route` was without any type parameters being provided. So it was always `Route<any, any>`.